### PR TITLE
Add support for seed data for entity types using table sharing where the principal is using TPT

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Sort table mappings consistently
Remove discriminator-related facets completely when using TPT
Simplify SharedTableEntryMap

Part of #2266

### CODE REVIEW

File: src/EFCore.Relational/Metadata/Internal/TableMappingBaseComparer.cs [hunk 1]
```diff
@@ -48,8 +48,8 @@ private TableMappingBaseComparer(bool forEntityType)
         public int Compare(ITableMappingBase x, ITableMappingBase y)
         {
             var result = _isForEntityType
-                ? x.IsMainTableMapping.CompareTo(y.IsMainTableMapping)
-                : x.IsMainEntityTypeMapping.CompareTo(y.IsMainEntityTypeMapping);
+                ? y.IsMainTableMapping.CompareTo(x.IsMainTableMapping)
```
Comment (smitpatel) [inline src/EFCore.Relational/Metadata/Internal/TableMappingBaseComparer.cs:51]: Is this right?
A quick check on int revealed that `a.CompareTo(b)` is same as `Comparer<int>.Default.Compare(a, b)`

Comment (AndriySvyryd) [reply src/EFCore.Relational/Metadata/Internal/TableMappingBaseComparer.cs:51]: What's your point? Would you prefer `Comparer<bool>.Default.Compare(y.IsMainTableMapping, x.IsMainTableMapping)`?

Comment (smitpatel) [reply src/EFCore.Relational/Metadata/Internal/TableMappingBaseComparer.cs:51]: `x` & `y` are inverted?

Comment (AndriySvyryd) [reply src/EFCore.Relational/Metadata/Internal/TableMappingBaseComparer.cs:51]: Yes, true should be sorted first, this was a bug

**Pull request merged**